### PR TITLE
decompile starter chooser

### PR DESCRIPTION
**Pull request by camthesaxman**
This is the screen with the three Pokeballs in which the player chooses their starter Pokemon. I'm having trouble getting sub_810A42C to match, but all the rest of the code is fine. Despite having curly braces around those statements, CSE still takes effect, and the compiler just increments the pointer each time instead of recalculating the address.

### CODE REVIEW

File: data/data2.s [hunk 1]
```diff
@@ -23919,15 +23919,22 @@ gBirchBallarrow_Gfx::
...
-
+	.byte 0x00, 0x09
+	.byte 0x10, 0x0A
+	.byte 0x08, 0x04
+
+gStarterMons:: @ 83F76C4
+	.2byte 277
+	.2byte 280
+	.2byte 283
+	.2byte 0
```
Comment (yenatch) [inline data/data2.s]: Why not use `SPECIES_TREECKO` `SPECIES_TORCHIC` `SPECIES_MUDKIP`?

**Comment by YamaArashi** [comment]
Doing this prevents the optimization, and puts the load of the "5" value in the right place, but the registers still don't match unfortunately.
```
    {
        u8 val = 0x5;
        {
            u8 *p = arr;
            p += 5;
            *p = 0xFC;
        }
        {
            u8 *p = arr;
            p += 6;
            *p = 0x11;
        }
        {
            u8 *p = arr;
            p += 7;
            *p = val;
        }
    }
```

File: data/data2.s (hunk 1 above)
Comment (camthesaxman) [inline data/data2.s]: I didn't know we had defined constants for those. Thanks, I'll add that.

**Comment by camthesaxman** [comment]
The code now compiles and matches the original rom. I just have to clean it up now.

**Comment by camthesaxman** [comment]
Should be ready to merge, now.

**Pull request merged**